### PR TITLE
Fixed MongoDB Authentication

### PR DESCRIPTION
**Pull request by madhawa-gunasekara**
Please don't merge till we release mongo db-java 3.0 orbit bundle

### CODE REVIEW

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/config/MongoConfig.java [hunk 1]
```diff
@@ -156,6 +161,32 @@ public MongoClient getMongoClient() {
...
     }
 
+    private MongoCredential createCredential(Map<String, String> properties) {
+        MongoCredential credential = null;
+        String authenticationType = properties.get(DBConstants.MongoDB.AUTHENTICATION_TYPE);
+        String username = properties.get(DBConstants.MongoDB.USERNAME);
+        String password = properties.get(DBConstants.MongoDB.PASSWORD);
+        String database = properties.get(DBConstants.MongoDB.DATABASE);
+        switch (authenticationType) {
+            case "PLAIN":
```
Comment (lafernando) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/config/MongoConfig.java]: Create string constants for the authentication types and use them here.

Comment (madhawa-gunasekara) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/config/MongoConfig.java]: @lafernando  Shall I create string constants in DBConstants in common package

Comment (lafernando) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/config/MongoConfig.java]: Yeah, that's fine.

Comment (lafernando) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/config/MongoConfig.java]: This is also shown here to have conflicts, please resolve them as well.

**Pull request merged**